### PR TITLE
Socketsetup

### PR DESCRIPTION
**Pull request by kevinzhangftw**

### CODE REVIEW

File: src/actions/SocketAction.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const socketResponse = (response) => dispatch => {
+  dispatch({
+    type: 'SOCKET_RESPONSE_RECEIVE',
```
Comment (kimjune01) [inline src/actions/SocketAction.js]: should be VERB_STUFF, not STUFF_VERB

File: src/containers/SocketContainer.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import { connect } from 'react-redux';
+
+const Socket = createSocket({
+  Component,
+  createElement
+});
+
+class SocketContainer extends Component {
+    // Socket internal state
+    state = {
```
Comment (kimjune01) [inline src/containers/SocketContainer.js:13]: get rid of local state please!

File: src/containers/SocketContainer.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        
+    };
+
+    onReconnect = ev => {
+        console.log("> Reconnecting...", ev);
+    };
+
+    sendMessage = _ => {
+        // WebSocket available in state!
+        const request = { "speech": this.props.text, "endpoint": "https://hap2a5df4m.execute-api.us-east-1.amazonaws.com/dev/ping", "state": { "directory": "home" } }
```
Comment (jyleong) [inline src/containers/SocketContainer.js:41]: We can think of moving some api endpoints to a service or at least a unified resource

File: src/containers/SocketContainer.js (hunk 2 above)
Comment (jyleong) [reply src/containers/SocketContainer.js:13]: reduced local state as much as we can. Must keep state.socket

**Pull request merged**